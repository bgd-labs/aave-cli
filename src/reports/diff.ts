/**
 * Highly inspired by - reimplemented a bit simpler & added types
 * @license MIT https://github.com/omgaz/diffler
 * Author: Gary Chisholm @omgaz
 */

export type Difference<
  A extends Record<string, any>,
  B extends Record<string, any>
> = { [key in keyof Omit<A, keyof B>]: { from: A[key]; to: null } } & {
  [key in keyof Omit<B, keyof A>]: { from: null; to: B[key] };
};

export type DiffOutput<
  A extends Record<string, any>,
  B extends Record<string, any>
> = {
  [key in keyof (A | B)]: DiffOutput<A[key], B[key]>;
} & Difference<A, B>;

export function diff<
  T extends Record<string, any>,
  X extends Record<string, any>
>(a: T, b: X, removeUnchanged?: boolean): DiffOutput<T, X> {
  const out = {} as Record<string, any>;
  for (const key in a) {
    if (!b.hasOwnProperty(key)) {
      out[key] = { from: a[key], to: null };
    } else {
      if (typeof a[key] === "object") {
        const tempDiff = diff(a[key], b[key], removeUnchanged);
        if (Object.keys(tempDiff).length > 0) {
          out[key] = tempDiff;
        }
      } else {
        if (b[key as string] == a[key]) {
          if (!removeUnchanged) out[key] = a[key];
        } else {
          out[key] = { from: a[key], to: b[key] };
        }
      }
    }
  }
  for (const key in b) {
    if (a.hasOwnProperty(key)) continue;
    out[key] = { from: null, to: b[key] };
  }
  return out as DiffOutput<T, X>;
}
