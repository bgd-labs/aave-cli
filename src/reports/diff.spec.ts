import { describe, it, expect } from "vitest";
import pre from "../mocks/preTestEngineArbV3.json";
import post from "../mocks/postTestEngineArbV3.json";
import { diff } from "./diff";

describe("diff", () => {
  it("should return same object if no changes were found", () => {
    const input = { a: "a" };
    const out = diff(input, { ...input });
    expect(JSON.stringify(input)).eq(JSON.stringify(out));
  });

  it("should return same object if no changes were found (nested)", () => {
    const input = { a: { a: "a" } };
    const out = diff(input, { ...input });
    expect(JSON.stringify(input)).eq(JSON.stringify(out));
  });

  it("should return same object if no changes were found (nested)", () => {
    const input0 = { a: { a: "a" } } as const;
    const input1 = { a: { a: "b" } } as const;
    const out = diff(input0, input1);
    expect(out.a.a).eql({ from: "a", to: "b" });
  });

  it("should find all the changes", () => {
    const out = diff(pre, post, true);
    console.log(JSON.stringify(out, null, 2));
  });
});
