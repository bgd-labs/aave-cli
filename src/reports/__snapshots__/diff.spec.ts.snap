// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`diff > should find all the changes 1`] = `
{
  "reserves": {
    "0x0b3F868E0BE5597D5DB7fEB59E1CADBb0fdDa50a": {
      "currentLiquidityRate": {
        "from": 1.3415266124935912e+22,
        "to": 1.3515266124935913e+22,
      },
      "currentVariableBorrowRate": {
        "from": 1.6150925573283835e+24,
        "to": 1.7150925573283834e+24,
      },
      "liquidityIndex": {
        "from": 1.0030723628446084e+27,
        "to": 1.1020723628446084e+27,
      },
      "variableBorrowIndex": {
        "from": 1.0191200042968116e+27,
        "to": 1.0211200042968116e+27,
      },
    },
    "0x385Eeac5cB85A38A9a07A70c73e0a3271CfB54A7": {
      "reserveFactor": {
        "from": 2000,
        "to": 3500,
      },
    },
    "0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619": {
      "interestRateStrategy": {
        "from": "0x03733F4E008d36f2e37F0080fF1c8DF756622E6F",
        "to": "0x27eFE5db315b71753b2a38ED3d5dd7E9362ba93F",
      },
      "reserveFactor": {
        "from": 1000,
        "to": 1500,
      },
    },
    "0x85955046DF4668e1DD369D2DE9f3AEB98DD2A369": {
      "reserveFactor": {
        "from": 2000,
        "to": 3500,
      },
    },
    "0xE0B52e49357Fd4DAf2c15e02058DCE6BC0057db4": {
      "interestRateStrategy": {
        "from": "0x41B66b4b6b4c9dab039d96528D1b88f7BAF8C5A4",
        "to": "0xA9F3C3caE095527061e6d270DBE163693e6fda9D",
      },
    },
    "0xE111178A87A3BFf0c8d18DECBa5798827539Ae99": {
      "interestRateStrategy": {
        "from": "0x41B66b4b6b4c9dab039d96528D1b88f7BAF8C5A4",
        "to": "0xA9F3C3caE095527061e6d270DBE163693e6fda9D",
      },
    },
    "0xa3Fa99A148fA48D14Ed51d610c367C61876997F1": {
      "interestRateStrategy": {
        "from": "0x41B66b4b6b4c9dab039d96528D1b88f7BAF8C5A4",
        "to": "0xA9F3C3caE095527061e6d270DBE163693e6fda9D",
      },
      "reserveFactor": {
        "from": 1000,
        "to": 2000,
      },
    },
    "0xc2132D05D31c914a87C6611C10748AEb04B58e8F": {
      "interestRateStrategy": {
        "from": "0x41B66b4b6b4c9dab039d96528D1b88f7BAF8C5A4",
        "to": "0xA9F3C3caE095527061e6d270DBE163693e6fda9D",
      },
    },
  },
  "strategies": {
    "0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619": {
      "address": {
        "from": "0x03733F4E008d36f2e37F0080fF1c8DF756622E6F",
        "to": "0x27eFE5db315b71753b2a38ED3d5dd7E9362ba93F",
      },
      "baseStableBorrowRate": {
        "from": 9e+25,
        "to": 6.8e+25,
      },
      "baseVariableBorrowRate": {
        "from": 0,
        "to": 1e+25,
      },
      "maxExcessUsageRatio": {
        "from": 5.5e+26,
        "to": 2e+26,
      },
      "optimalUsageRatio": {
        "from": 4.5e+26,
        "to": 8e+26,
      },
      "stableRateSlope1": {
        "from": 0,
        "to": 4e+25,
      },
      "stableRateSlope2": {
        "from": 0,
        "to": 8e+26,
      },
      "variableRateSlope1": {
        "from": 7e+25,
        "to": 3.8e+25,
      },
      "variableRateSlope2": {
        "from": 3e+27,
        "to": 8e+26,
      },
    },
    "0xE0B52e49357Fd4DAf2c15e02058DCE6BC0057db4": {
      "address": {
        "from": "0x41B66b4b6b4c9dab039d96528D1b88f7BAF8C5A4",
        "to": "0xA9F3C3caE095527061e6d270DBE163693e6fda9D",
      },
      "maxExcessUsageRatio": {
        "from": 1e+26,
        "to": 2e+26,
      },
      "optimalUsageRatio": {
        "from": 9e+26,
        "to": 8e+26,
      },
      "stableRateSlope2": {
        "from": 6e+26,
        "to": 7.5e+26,
      },
      "variableRateSlope2": {
        "from": 6e+26,
        "to": 7.5e+26,
      },
    },
    "0xE111178A87A3BFf0c8d18DECBa5798827539Ae99": {
      "address": {
        "from": "0x41B66b4b6b4c9dab039d96528D1b88f7BAF8C5A4",
        "to": "0xA9F3C3caE095527061e6d270DBE163693e6fda9D",
      },
      "maxExcessUsageRatio": {
        "from": 1e+26,
        "to": 2e+26,
      },
      "optimalUsageRatio": {
        "from": 9e+26,
        "to": 8e+26,
      },
      "stableRateSlope2": {
        "from": 6e+26,
        "to": 7.5e+26,
      },
      "variableRateSlope2": {
        "from": 6e+26,
        "to": 7.5e+26,
      },
    },
    "0xa3Fa99A148fA48D14Ed51d610c367C61876997F1": {
      "address": {
        "from": "0x41B66b4b6b4c9dab039d96528D1b88f7BAF8C5A4",
        "to": "0xA9F3C3caE095527061e6d270DBE163693e6fda9D",
      },
      "maxExcessUsageRatio": {
        "from": 1e+26,
        "to": 2e+26,
      },
      "optimalUsageRatio": {
        "from": 9e+26,
        "to": 8e+26,
      },
      "stableRateSlope2": {
        "from": 6e+26,
        "to": 7.5e+26,
      },
      "variableRateSlope2": {
        "from": 6e+26,
        "to": 7.5e+26,
      },
    },
    "0xc2132D05D31c914a87C6611C10748AEb04B58e8F": {
      "address": {
        "from": "0x41B66b4b6b4c9dab039d96528D1b88f7BAF8C5A4",
        "to": "0xA9F3C3caE095527061e6d270DBE163693e6fda9D",
      },
      "maxExcessUsageRatio": {
        "from": 1e+26,
        "to": 2e+26,
      },
      "optimalUsageRatio": {
        "from": 9e+26,
        "to": 8e+26,
      },
      "stableRateSlope2": {
        "from": 6e+26,
        "to": 7.5e+26,
      },
      "variableRateSlope2": {
        "from": 6e+26,
        "to": 7.5e+26,
      },
    },
  },
}
`;

exports[`diff > should find all the changes in 3.1 1`] = `
{
  "reserves": {
    "0x3C8Ca53ee5661D29d3d3C0732689a4b86947EAF0": {
      "address": {
        "from": null,
        "to": "0xF9514C867b37D36a5affB003582b1818dA3870eb",
      },
      "baseVariableBorrowRate": {
        "from": null,
        "to": "0",
      },
      "maxVariableBorrowRate": {
        "from": null,
        "to": "640000000000000000000000000",
      },
      "optimalUsageRatio": {
        "from": null,
        "to": "450000000000000000000000000",
      },
      "variableRateSlope1": {
        "from": null,
        "to": "40000000000000000000000000",
      },
      "variableRateSlope2": {
        "from": null,
        "to": "600000000000000000000000000",
      },
    },
    "0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f": {
      "address": {
        "from": null,
        "to": "0xF9514C867b37D36a5affB003582b1818dA3870eb",
      },
      "baseVariableBorrowRate": {
        "from": null,
        "to": "0",
      },
      "maxVariableBorrowRate": {
        "from": null,
        "to": "640000000000000000000000000",
      },
      "optimalUsageRatio": {
        "from": null,
        "to": "450000000000000000000000000",
      },
      "variableRateSlope1": {
        "from": null,
        "to": "40000000000000000000000000",
      },
      "variableRateSlope2": {
        "from": null,
        "to": "600000000000000000000000000",
      },
    },
    "0xffD4505B3452Dc22f8473616d50503bA9E1710Ac": {
      "address": {
        "from": null,
        "to": "0xF9514C867b37D36a5affB003582b1818dA3870eb",
      },
      "baseVariableBorrowRate": {
        "from": null,
        "to": "0",
      },
      "liquidationBonus": {
        "from": 10500,
        "to": 11100,
      },
      "liquidationThreshold": {
        "from": 8600,
        "to": 9000,
      },
      "ltv": {
        "from": 8250,
        "to": 6200,
      },
      "maxVariableBorrowRate": {
        "from": null,
        "to": "640000000000000000000000000",
      },
      "optimalUsageRatio": {
        "from": null,
        "to": "450000000000000000000000000",
      },
      "variableRateSlope1": {
        "from": null,
        "to": "40000000000000000000000000",
      },
      "variableRateSlope2": {
        "from": null,
        "to": "600000000000000000000000000",
      },
    },
  },
}
`;

exports[`diff > should find all the changes in 3.2 1`] = `
{
  "eModes": {
    "1": {
      "borrowableBitmap": {
        "from": null,
        "to": "1171493",
      },
      "collateralBitmap": {
        "from": null,
        "to": "1171493",
      },
    },
    "2": {
      "borrowableBitmap": {
        "from": null,
        "to": "393344",
      },
      "collateralBitmap": {
        "from": null,
        "to": "393344",
      },
    },
    "3": {
      "borrowableBitmap": {
        "from": null,
        "to": "524304",
      },
      "collateralBitmap": {
        "from": null,
        "to": "524304",
      },
    },
  },
  "poolConfig": {
    "poolConfiguratorImpl": {
      "from": "0x419226e0Ad27f3B2019123f7246a364622b018e5",
      "to": "0xc6988946e3E490F3E21F8647Db2b81fD4D014acA",
    },
    "poolImpl": {
      "from": "0xc4F7b5d4ca00eE04cF9887D5D811d3C5d6506477",
      "to": "0xf877225774FBf56Db46DB017Ba9455CA64EF4e32",
    },
    "protocolDataProvider": {
      "from": "0x7deEB8aCE4220643D8edeC871a23807E4d006eE5",
      "to": "0x21d35DBF31bdfFbe40996C78Acae2058682B0774",
    },
  },
  "reserves": {
    "0x03b54A6e9a984069379fae1a4fC4dBAE93B3bCCD": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x0b3F868E0BE5597D5DB7fEB59E1CADBb0fdDa50a": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x172370d5Cd63279eFa6d502DAB29171933a610AF": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x1BFD67037B42Cf73acF2047067bd4F2C47D9BfD6": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x385Eeac5cB85A38A9a07A70c73e0a3271CfB54A7": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x3A58a54C066FdC0f2D55FC9C89F0415C92eBf3C4": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x3c499c542cEF5E3811e1192ce70d8cC03d5c3359": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x4e3Decbb3645551B8A19f0eA1678079FCB33fB4c": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x53E0bca35eC356BD5ddDFebbD1Fc0fD03FaBad39": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x85955046DF4668e1DD369D2DE9f3AEB98DD2A369": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x8f3Cf7ad23Cd3CaDbD9735AFf958023239c6A063": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x9a71012B13CA4d3D0Cdc72A177DF3ef03b0E76A3": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xD6DF932A45C0f255f85145f286eA0b292B21C90B": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xE0B52e49357Fd4DAf2c15e02058DCE6BC0057db4": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xE111178A87A3BFf0c8d18DECBa5798827539Ae99": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xa3Fa99A148fA48D14Ed51d610c367C61876997F1": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xc2132D05D31c914a87C6611C10748AEb04B58e8F": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xfa68FB4628DFF1028CFEc22b4162FCcd0d45efb6": {
      "interestRateStrategy": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
  },
  "strategies": {
    "0x03b54A6e9a984069379fae1a4fC4dBAE93B3bCCD": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x0b3F868E0BE5597D5DB7fEB59E1CADBb0fdDa50a": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x172370d5Cd63279eFa6d502DAB29171933a610AF": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x1BFD67037B42Cf73acF2047067bd4F2C47D9BfD6": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x385Eeac5cB85A38A9a07A70c73e0a3271CfB54A7": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x3A58a54C066FdC0f2D55FC9C89F0415C92eBf3C4": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x3c499c542cEF5E3811e1192ce70d8cC03d5c3359": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x4e3Decbb3645551B8A19f0eA1678079FCB33fB4c": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x53E0bca35eC356BD5ddDFebbD1Fc0fD03FaBad39": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x85955046DF4668e1DD369D2DE9f3AEB98DD2A369": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x8f3Cf7ad23Cd3CaDbD9735AFf958023239c6A063": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0x9a71012B13CA4d3D0Cdc72A177DF3ef03b0E76A3": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xD6DF932A45C0f255f85145f286eA0b292B21C90B": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xE0B52e49357Fd4DAf2c15e02058DCE6BC0057db4": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xE111178A87A3BFf0c8d18DECBa5798827539Ae99": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xa3Fa99A148fA48D14Ed51d610c367C61876997F1": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xc2132D05D31c914a87C6611C10748AEb04B58e8F": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
    "0xfa68FB4628DFF1028CFEc22b4162FCcd0d45efb6": {
      "address": {
        "from": "0x21128C8f5AE37904c9082D932928FE0C2be1Ff73",
        "to": "0x56076f960980d453b5B749CB6A1c4D2E4e138B1A",
      },
    },
  },
}
`;
